okhttp/src/main/java/okhttp3/internal/Platform.java

```diff
@@ -123,6 +126,16 @@ public void log(String message) {
     System.out.println(message);
   }
 
+  public static List<String> alpnProtocolNames(List<Protocol> protocols) {
+    List<String> names = new ArrayList<>(protocols.size());
+    for (int i = 0, size = protocols.size(); i < size; i++) {
```

Comment:
it is from https://speakerdeck.com/jakewharton/eliminating-code-overhead-square-hq-2015